### ISSUE TITLE
StreamPowerEroder gets stuck when n_sp != 1

### ISSUE THREAD

**Issue by awickert** [landlab/landlab#473]
I've set up the fluvial component of the model as:
```
StreamPowerEroder(mg, K_sp=K_sp, m_sp=m_sp, n_sp=n_sp, threshold_sp=0, use_Q=None)
```

I have then run it within a "while" loop with time-steps every 10 years.

It solves when n_sp = 1, but never finishes the first time step when n_sp < 1. If more code is needed, I'll clean up an example version and post it.

I am using the main development branch, updated as of the time this was posted.

**Comment by SiccarPoint** [comment]
Hi Andy,

This is a known issue - #444, #431. It has been resolved by #433, at least, into our `master` repo. However, it's entirely possible we've failed to make a release version with this.

If you're maintaining a dev version, update it and everything should be good. (Remember to recompile the binaries as necessary: `python setup.py develop`)

If you're running a release, confirm you've updated it since late Feb. If you have, drop us another line and we'll ensure we get a new release out with the fix ASAP.

**Comment by SiccarPoint** [comment]
Resolved per Andy's comment on #444

**Comment by nicgaspar** [comment]
@SiccarPoint I think this issue might still be here or something else is coming up. I have the latest version of the code. 

Today I tried to run stream power with this setup:

sp = FastscapeEroder(mg, K_sp=0.00005, m_sp=0.4, n_sp=0.8, threshold_sp=0.,
                    rainfall_intensity=1.)

And I get this error:

  File "/Users/nicgaspar/landlab/landlab/components/stream_power/fastscape_stream_power.py", line 382, in erode
    threshdt, alpha_divided, n, z)

  File "landlab/components/stream_power/cfuncs.pyx", line 185, in landlab.components.stream_power.cfuncs.erode_with_link_alpha_fixthresh (landlab/components/stream_power/cfuncs.c:3142)
    assert niter < 50, 'failure to converge in SP solver'

AssertionError: failure to converge in SP solver

More information is that @gregtucker added the line in which the code is failing on January 22, 2017. So I think this new problem may be related to the fix for issue #431 

Also note that the code appears to run (maybe a bit slowly, but not crashing) if we just comment out line 185 in landlab/components/stream_power/cfuncs.pyx.
